objstorage/objstorageprovider/provider.go

```diff
@@ -301,8 +308,39 @@ func (p *provider) LinkOrCopyFromLocal(
 		p.addMetadata(meta)
 		return meta, nil
 	}
-	// TODO(radu): for the copy case, we should use `p.Create` and do the copy ourselves.
-	panic("unimplemented")
+	// Create the object and copy the data.
+	w, meta, err := p.Create(ctx, dstFileType, dstFileNum, opts)
+	if err != nil {
+		return objstorage.ObjectMetadata{}, err
+	}
+	f, err := srcFS.Open(srcFilePath, vfs.SequentialReadsOption)
+	if err != nil {
+		return objstorage.ObjectMetadata{}, err
+	}
+	defer f.Close()
+	buf := make([]byte, 64*1024)
```

Comment:
Just wondering, any particular reason to do a copy manually manually instead of io.Copy() or io.CopyBuffer() ?

Reply:
We're not an `io.Writer`